matching-core/src/main/java/com/graphhopper/matching/MatchSequence.java

```diff
@@ -0,0 +1,323 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for 
+ *  additional information regarding copyright ownership.
+ * 
+ *  GraphHopper GmbH licenses this file to you under the Apache License, 
+ *  Version 2.0 (the "License"); you may not use this file except in 
+ *  compliance with the License. You may obtain a copy of the License at
+ * 
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.matching;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.bmw.hmm.SequenceState;
+import com.graphhopper.routing.Path;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.EdgeIteratorState;
+
+/**
+ * Used to represent a continuous map matching sequence.
+ * 
+ * @author kodonnell
+ */
+public class MatchSequence {
+    /**
+     * Describing the reason for the sequence to have ended:
+     *      - UNKNOWN: we don't know why it ended
+     *      - LAST_GPX_ENTRY: it was the last in the GPX track, so had to end.
+     *      - NO_CANDIDATES: there were no candidates for the *next* step
+     *      - NO_POSSIBLE_TRANSITIONS: there were no possible transitions to the *next* step.
+     */
+    public static enum ViterbiBreakReason { UNKNOWN, LAST_GPX_ENTRY, NO_CANDIDATES, NO_POSSIBLE_TRANSITIONS };
+    /**
+     * Sequence type descriptor:
+     *      - STATIONARY: where there was only a single step in the sequence. 
+     *      - SEQUENCE: a 'normal' sequence, i.e. anything not covered by the above.
+     */
+    public static enum SequenceType { SEQUENCE, STATIONARY, UNKNOWN };
+    /**
+     * The break reason for this sequence.
+     */
+    public final ViterbiBreakReason viterbiBreakReason;
+    /**
+     * The sequence type.
+     */
+    public final SequenceType type;
+    /**
+     * The matched sequence, as returned from viterbi.computeMostLikelySequence().
+     */
+    public final List<SequenceState<Candidate, MatchEntry, Path>> matchedSequence;
```

Comment:
I could just make it private? The user information (edges, original points in sequence, etc.) can be obtained by the user without it.